### ISSUE TITLE
Functions in rootisolation.py should assume rational bounds for intervals/rectangles

### ISSUE THREAD

**Issue by skirpichev** [sympy/sympy#14804]
Consider [L301](https://github.com/sympy/sympy/blob/eb6382a1cdd1a87f7ef950dc588923273d2afa0f/sympy/polys/rootisolation.py#L301): here K.zero/K.one are used to initialize boundaries for interval, isolating a real root.  Currently, K can be only ZZ, functions don't accept anything else, see e.g. [L503](https://github.com/sympy/sympy/blob/eb6382a1cdd1a87f7ef950dc588923273d2afa0f/sympy/polys/rootisolation.py#L503).  But it seems we can use same code for more generic situations, namely for polynomials over real algebraic fields (see PR diofant/diofant#630).  And that's important: only rational operations are made on boundaries of the root interval.  In short: if we start from rational boundaries - we will end with rational boundaries.

Does it make any sense to support generic (from the domain of coefficients for our polynomial) end points for the isolation interval?

**Comment by jksuom** [comment]
It seems to me that the isolation code should work over any ordered field K. Currently, Q is the only choice but K could also be any real algebraic field provided its order is properly implemented. For such a field K, I think it would make perfect sense to support endpoints belonging to K for isolation intervals as they arise naturally in the process. Rational endpoints could probably be also found if necessary, but that would mean extra work.

**Comment by skirpichev** [comment]
> It seems to me that the isolation code should work over any ordered field K.

Yes, it should.  At least with some minor modifications (see PR).

@jksuom, the question: is it worth supporting such endpoints?

> as they arise naturally in the process

In fact, they don't arise in the process, that's the point, unless specified with sup/inf parameters.  If they are not specified - end points should stay rational.  See example from [this commit message](https://github.com/diofant/diofant/pull/630/commits/2ef4c01311d3b408706f84f59ba7815014d2f704).

**Comment by jksuom** [comment]
Hmm... I forget the details, but if it is easy to find intervals with rational endpoints, then they should probably be preferred. Could non-rational endpoints be useful in some connection?

**Comment by skirpichev** [comment]
> Could non-rational endpoints be useful in some connection?

I can't suggest use case.

**Comment by jksuom** [comment]
I cannot think of it either.
